### DISCUSSION TITLE
Make the `pos` member of `Node` public

### DISCUSSION THREAD

**Discussion by sherpal** [VirtusLab/scala-yaml#272] [Ideas]
I am opening this discussion to see if it would be possible to open the [`pos` member](https://github.com/VirtusLab/scala-yaml/blob/main/core/shared/src/main/scala/org/virtuslab/yaml/Node.scala#L12) of the `Node` trait and make it public.

Being an immutable variable, I don't see a reason why it's currently private, and being able to track back a node origin can be valuable.

**Comment by roman-mibex-2** [comment]
Our use case for extended error message and give users of your system more information:

- The YAML is parsed correctly. But later down the line an issue arise. We would love to report the error back to the user with the exact YAML like.
   Examples:
     - The YAML has some URL configured. Later on in the system the URLs turned  can't be accessed. We want to report where that URL was configured in the YAML
     - We have a list of things, each with a name. The name has to be unique. We want to report that name as invalid and point to the exact location.  The name needs to unique accros files, so only when everything is gathered we can report the issue.
     - Everything is correct per see. But we want to show on the UI why something behaves like it does. For that we want to be able to show the user the configuration line in YAML file.

PS: Currently we hack it by placing an extra class into the `org.virtuslab.yaml` package.

**Comment by tgodzik** [comment]
I think that's perfectly fine, I don't know the initial reason for making it private, but since this doesn't break anything, I think it's perfectly fine.

Unless @kpodsiad @lwronski have some more background here?
